### PR TITLE
Implement Azure benchmark

### PR DESCRIPTION
**Pull request by jeniawhite**
### Summary of your changes
Implementation of the Azure benchmark flavor.
This is supposed to be merged after all of the other related Azure PRs in order to finalize the first drop.

**NOTE: 
Please merge after all related PRs**

### Related Issues
- Fixes: #1290 
- Related: #1307 
- Related: #1308 
- Related: #1309 
- Related: #1310 

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: flavors/benchmark/azure.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	"github.com/elastic/cloudbeat/dataprovider"
+	"github.com/elastic/cloudbeat/dataprovider/providers/cloud"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/fetching/factory"
+	"github.com/elastic/cloudbeat/resources/fetching/registry"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/auth"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
+)
+
+type AZURE struct {
```
Comment (orestisfl) [inline flavors/benchmark/azure.go:37]:
```suggestion
type Azure struct {
```

No need to be all caps, since Azure is not an initialism

**Comment by jeniawhite** [comment]
**NOTE:
Please merge after all related PRs**

File: flavors/benchmark/azure.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	azureConfig, err := a.CfgProvider.GetAzureClientConfig()
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to initialize azure config: %w", err)
+	}
+
+	// azureIdentity, err := a.IdentityProvider.GetIdentity(ctx, azureConfig)
+	// if err != nil {
+	// 	return nil, nil, fmt.Errorf("failed to get Azure identity: %v", err)
+	// }
+
```
Comment (orestisfl) [inline flavors/benchmark/azure.go:72]:
```suggestion
```

File: flavors/benchmark/azure.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		cloud.NewDataProvider(cloud.WithLogger(log)),
+		nil,
+		nil
+}
+
+func (a *Azure) checkDependencies() error {
+	// if a.IdentityProvider == nil {
+	// 	return errors.New("azure identity provider is uninitialized")
+	// }
+
```
Comment (orestisfl) [inline flavors/benchmark/azure.go:93]:
```suggestion
```

File: flavors/benchmark/azure.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/elastic/cloudbeat/flavors/benchmark/builder"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/fetching/preset"
+	"github.com/elastic/cloudbeat/resources/fetching/registry"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/auth"
+	"github.com/elastic/cloudbeat/resources/providers/azurelib/inventory"
+)
+
+type Azure struct {
+	// IdentityProvider     identity.ProviderGetter
```
Comment (orestisfl) [inline flavors/benchmark/azure.go:39]:
```suggestion
```

**Review by orestisfl — COMMENTED** [review]
Let's remove the commented-out parts for now (commit can be reverted locally later)

**Review by orestisfl — APPROVED** [review]

**Pull request merged**